File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    int countGoodNodes(TreeNode* root, int biggestValUpstream){
        if(!root){
            return 0;
        }
        int num = root->val;
        if(num >= biggestValUpstream){
            return 1 + countGoodNodes(root->left, num) + countGoodNodes(root->right, num);
        }
        return countGoodNodes(root->left, biggestValUpstream) + countGoodNodes(root->right, biggestValUpstream);
    }

    int goodNodes(TreeNode* root) {
        return countGoodNodes(root, -101);
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers) - 1
        while numbers[left] + numbers[right] != target:
            sm = numbers[left] + numbers[right]
            if sm < target:
                left += 1
            else:
                right -= 1
        
        return [left + 1, right + 1]